test/e2e/apps/job.go

```diff
@@ -41,7 +42,7 @@ var _ = SIGDescribe("Job", func() {
 	// Simplest case: N pods succeed
 	ginkgo.It("should run a job to completion when tasks succeed", func() {
 		ginkgo.By("Creating a job")
-		job := jobutil.NewTestJob("succeed", "all-succeed", v1.RestartPolicyNever, parallelism, completions, nil, backoffLimit)
+		job := jobutil.NewTestJob("succeed", "all-succeed", v1.RestartPolicyNever, parallelism, completions, nil, backoffLimit, "")
```

Comment:
Nit: I think adding the `""` as the final argument harms the readability of `NewTestJob`. For someone seeing just the function call, its not clear what the purpose of `""` is.

I think there are a couple of options:

1. We could give `""` a descriptive variable name... for example:
```
runOnAnyNode := ""
job := jobutil.NewTestJob(...", runOnAnyNode)
```
2. We could define two separate functions - one for running a function on a random node and one for testing a job on a dedicated node. Then, only the tests which actually need to specify a specific node would need to concern themselves with this parameter. These separate functions could share code to decrease duplication.

Reply:
Nice suggestion, I applied the second one, now only necessary invocation is affected.